semantic-scope-graph/src/Stack/Path.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedLists #-}
+module Stack.Path
+  ( Path (..)
+  , Edge (..)
+  , StartingSize (..)
+  , PathInvariantError (..)
+  , checkEdgeInvariants
+  , checkNodeInvariants
+  , Validity (..)
+  , validity
+  , Completion (..)
+  , completion
+  , isIncremental
+  ) where
+
+
+import Data.Functor.Tagged
+import Data.Monoid
+import Data.Text (Text)
+import Stack.Graph (Node (..), Symbol)
+import Data.Semigroup (sconcat)
+import Data.Sequence (Seq (..))
+
+data Path = Path
```

Comment:
Could we add some 📝 to this? Is the intent to generate `Path` datatypes from the stack graph, then check their validity?

Reply:
I think we want to do a lot more than check their validity, but I wanted to run this design by y’all before we spec out more functionality on the Haskell side.